src/Resources/app/administration/src/module/swag-migration/service/swag-migration-error-resolution.service.ts

```diff
@@ -0,0 +1,708 @@
+import type { Property } from '@administration/src/core/data/entity-definition.data';
+import type EntityDefinition from '@administration/src/core/data/entity-definition.data';
+
+/**
+ * @private
+ * structure holding categorized entity fields.
+ */
+export interface EntityFields {
+    scalar: Record<string, Property>;
+    associations: Record<string, Property>;
+    required: Record<string, Property>;
+}
+
+/**
+ * @private
+ * table column definition for error resolution modal.
+ */
+export interface TableColumn {
+    label: string;
+    property: string;
+    sortable: boolean;
+    position: number;
+    visible?: boolean;
+}
+
+/**
+ * @private
+ * data types used in entity definitions.
+ * `scalarTypes` & `jsonTypes` in `entity-definition.data.ts`
+ */
+export const DATA_TYPES = {
+    UUID: 'uuid',
+    INT: 'int',
+    TEXT: 'text',
+    FLOAT: 'float',
+    STRING: 'string',
+    BOOLEAN: 'boolean',
+    DATE: 'date',
+    JSON_LIST: 'json_list',
+    JSON_OBJECT: 'json_object',
+    ASSOCIATION: 'association',
+} as const;
+
+/**
+ * @private
+ * list of field types that are not handled for error resolution.
+ * blob and password fields are excluded due to their sensitive or complex nature.
+ */
+export const UNHANDLED_FIELD_TYPES = [
+    'blob',
+    'password',
+] as const;
+
+/**
+ * @private
+ * list of field names that are not handled for error resolution.
+ */
+export const UNHANDLED_FIELD_NAMES = [
+    'id',
+    'autoIncrement',
+    'translated',
+] as const;
+
+/**
+ * @private
+ * list of relation types that are handled for error resolution.
+ * `Property.relation` in `entity-definition.data.ts`
+ */
+export const HANDLED_RELATION_TYPES = {
+    MANY_TO_ONE: 'many_to_one',
+    ONE_TO_MANY: 'one_to_many',
+    MANY_TO_MANY: 'many_to_many',
+} as const;
+
+/**
+ * @private
+ * field component types used for rendering fields in the ui.
+ */
+export const FIELD_COMPONENT_TYPES = {
+    NUMBER: 'number',
+    TEXTAREA: 'textarea',
+    TEXT: 'text',
+    SWITCH: 'switch',
+    DATEPICKER: 'datepicker',
+    EDITOR: 'editor',
+} as const;
+
+/**
+ * @private
+ * mapping of data types to corresponding field component types for rendering in the ui.
+ */
+export const FIELD_TYPE_COMPONENT_MAPPING = {
+    [DATA_TYPES.INT]: FIELD_COMPONENT_TYPES.NUMBER,
+    [DATA_TYPES.TEXT]: FIELD_COMPONENT_TYPES.TEXTAREA,
+    [DATA_TYPES.FLOAT]: FIELD_COMPONENT_TYPES.NUMBER,
+    [DATA_TYPES.STRING]: FIELD_COMPONENT_TYPES.TEXT,
+    [DATA_TYPES.BOOLEAN]: FIELD_COMPONENT_TYPES.SWITCH,
+    [DATA_TYPES.DATE]: FIELD_COMPONENT_TYPES.DATEPICKER,
+    [DATA_TYPES.JSON_LIST]: FIELD_COMPONENT_TYPES.EDITOR,
+    [DATA_TYPES.JSON_OBJECT]: FIELD_COMPONENT_TYPES.EDITOR,
+} as const;
+
+/**
+ * @private
+ * list of fields prioritized for sorting purposes, to determined most meaningful fields first.
+ */
+export const PRIORITY_FIELDS = [
+    'id',
```

Comment:
small nitpick: I would move id further down, as we humans can't really read out much of UUIDs and can identifier data much better by descriptive names like `name`, `title`, `label`, `technicalName`, `firstName`, ...